src/OutputPresenter.ts

```diff
@@ -0,0 +1,268 @@
+import * as mobx from "mobx";
+import { templates } from "./Components/Template";
+import type {
+  Mood,
+  TrackEl,
+  LengthEl,
+  DurationTypes,
+} from "./Components/Template";
+
+const MAXLEN = 25;
+
+export class OutputPresenter {
+  constructor() {
+    mobx.makeObservable(this);
+  }
+  @mobx.observable
+  play: boolean = false;
+
+  // same as mediaCounter
+  @mobx.observable
+  playingMedia: number = 0;
+
+  @mobx.observable
+  playedSeconds: number = 0;
+
+  @mobx.observable
+  overallPlayedSeconds: number = 0;
+
+  @mobx.observable
+  currMood: Mood = templates[0];
+  @mobx.observable
+  trackIndex: number = 0;
+  @mobx.observable
+  currTrack: TrackEl = this.currMood.tracks[this.trackIndex];
+
+  @mobx.observable
+  lengthIndex: DurationTypes = "short";
+
+  @mobx.observable
+  currLength: LengthEl = this.currMood.tracks[0][this.lengthIndex];
+
+  @mobx.observable
+  musicLoaded: boolean = false;
+
+  @mobx.observable
+  music: HTMLAudioElement = new Audio(this.currTrack.musicTrack);
+
+  @mobx.computed
+  get isPlaying(): boolean {
+    return this.play;
+  }
+
+  @mobx.computed
+  get currMoodLen(): number {
+    return this.currMood.tracks.length;
+  }
+
+  @mobx.action
+  playVideo(): void {
+    if (this.playedSeconds >= this.currLength.totalDuration) {
+      console.log("resetting video");
+      this.resetVideo();
+    }
+    this.music.play();
+    this.play = true;
+  }
+  @mobx.action
+  pauseVideo(): void {
+    this.music.pause();
+    this.play = false;
+  }
+
+  @mobx.action
+  setPlayingMedia(newMedia: number): void {
+    this.playingMedia = newMedia;
+  }
+
+  @mobx.computed
+  get currPlayingMedia(): number {
+    return this.playingMedia;
+  }
+
+  @mobx.action
+  increPlayedSeconds(seconds: number) {
+    this.playedSeconds += seconds;
+    this.overallPlayedSeconds += seconds;
+    this.adjustSound();
+    if (this.playedSeconds >= this.currLength.slot[this.playingMedia]) {
+      this.playingMedia += 1;
+      this.playedSeconds = 0;
+    }
+  }
+
+  @mobx.action
+  adjustSound() {
+    if (this.music.currentTime >= this.currLength.end - 4) {
```

Comment:
why 4?
Best practise here for a constant would be to move it to the top of your file, like you did with `MAXLEN`

Reply:
the volume will decrement slowly at the last 4 seconds of the video to make the ending smoother :p